backend/core/src/Core.API/ResponseHandling/SignatureVerificationMiddleware.cs

```diff
@@ -0,0 +1,185 @@
+using System.Text;
+using static Core.Domain.Constants;
+
+namespace Core.API.ResponseHandling
+{
+    public class SignatureVerificationMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ILogger<SignatureVerificationMiddleware> _logger;
+
+        public SignatureVerificationMiddleware(
+            RequestDelegate next,
+            ILogger<SignatureVerificationMiddleware> logger)
+        {
+            _next = next;
+            _logger = logger;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            try
+            {
+                // Retrieve headers from the request
+                string? signatureHeader = context.Request.Headers["x-signature"];
+                string? payloadHeader = context.Request.Headers["x-payload"];
+
+                // Retrieve method from the request
+                var method = context.Request.Method;
+
+                if (string.IsNullOrWhiteSpace(payloadHeader))
+                {
+                    _logger.LogError("Missing payload header");
+                    var customErrors = new CustomErrors(new CustomError("Forbidden", "Missing Header", "x-payload"));
+                    await WriteCustomErrors(context.Response, customErrors, (int)HttpStatusCode.Forbidden);
+                    return;
+                }
+
+                if (string.IsNullOrWhiteSpace(signatureHeader))
+                {
+                    _logger.LogError("Missing x-signature header");
+                    var customErrors = new CustomErrors(new CustomError("Forbidden", "Missing Header", "x-signature"));
+                    await WriteCustomErrors(context.Response, customErrors, (int)HttpStatusCode.Forbidden);
+                    return;
+                }
+
+                byte[] payloadBytes = Encoding.UTF8.GetBytes(payloadHeader);
+
+                JObject payloadJson = JObject.Parse(payloadHeader);
+
+                string? publicKey = null;
+                string? postPayload = null;
+
+                if (method == "POST" || method == "PUT")
+                {
+                    // Check if the "PostPayload" property is present
+                    if (payloadJson.TryGetValue(SignaturePayload.PostPayload, out var post))
+                    {
+                        postPayload = (string?)post;
+                    }
+                    else
+                    {
+                        _logger.LogError("Missing postPayload header");
+                        var customErrors = new CustomErrors(new CustomError("Forbidden", "Missing Header", "postPayload"));
+                        await WriteCustomErrors(context.Response, customErrors, (int)HttpStatusCode.Forbidden);
+                        return;
+                    }
+                }
+
+                // Check if the "publicKey" property is present
+                if (payloadJson.TryGetValue(SignaturePayload.PublicKey, out var pubKey))
```

Comment:
should getting publickey be case sensitive; i.e. publicKey is allowed but publickey and PublicKey are not? (same for the other values)

Reply:
It is case-sensitive. So should be always what is defined in the constant file : publicKey